File: TransCoder_tokenized_test_set_functions/cpp/FIND_EXPRESSION_DUPLICATE_PARENTHESIS_NOT.cpp
bool findDuplicateparenthesis ( string str ) {
  stack < char > Stack;
  for ( char ch : str ) {
    if ( ch == ')' ) {
      char top = Stack . top ( );
      Stack . pop ( );
      int elementsInside = 0;
      while ( top != '(' ) {
        elementsInside ++;
        top = Stack . top ( );
        Stack . pop ( );
      }
      if ( elementsInside < 1 ) {
        return 1;
      }
    }
    else Stack . push ( ch );
  }
  return false;
}

File: results/java_codex/FIND_EXPRESSION_DUPLICATE_PARENTHESIS_NOT.java
bool findDuplicateparenthesis(string str)
{
    stack<char> Stack;
    for (int i = 0; i < str.length(); i++)
    {
        if (str[i] == ')')
        {
            char top = Stack.top();
            Stack.pop();
            int elementsInside = 0;
            while (top != '(')
            {
                elementsInside++;
                top = Stack.top();
                Stack.pop();
            }
            if (elementsInside < 1)
            {
                return true;
            }
        }
        else
        {
            Stack.push(str[i]);
        }
    }
    return false;
}